src/CMakeLists.txt

```diff
@@ -377,4 +378,28 @@ install (TARGETS ${PROJECT_NAME}
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
 )
-install (EXPORT ${PROJECT_NAME}-target DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME} FILE ${PROJECT_NAME}-config.cmake)
+
+# Generate and install CMake package
+include(CMakePackageConfigHelpers)
+set(PACKAGE_INSTALL_DESTINATION
+    ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}-${NN_PACKAGE_VERSION}
+)
+install(EXPORT ${PROJECT_NAME}-target
+    DESTINATION ${PACKAGE_INSTALL_DESTINATION}
+)
+write_basic_package_version_file(
+    ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config-version.cmake
+    VERSION ${NN_PACKAGE_VERSION}
+    COMPATIBILITY AnyNewerVersion
```

Comment:
This should be SameMajorVersion instead of AnyNewVersion I think.

Reply:
It is definitely your decision, what it should be, I am not familiar with the version semantics of nanomsg, but consider the following example:

The msgpack project also decided to use `SameMajorVersion` and if I want to express in my project, that I need version `2.1.5+` and that msgpack is required, I ended up writing this code:

```cmake
if(NOT msgpack_FOUND)
  find_package(msgpack 3.0.0)
  if(NOT msgpack_FOUND)
    find_package(msgpack 2.1.5 REQUIRED)
  endif()
endif()
```

You can see, that I will need to modify this code for every major release of msgpack.

While I agree with you that choosing `SameMajorVersion` is semantically more correct, I would recommend against it, because of the shortcomings of the `find_package` command on the user end.

Reply:
I stand by my request here.  MsgPack is a different project, and does not seem to follow Semantic Versioning.  We do, and are very careful to avoid bumping the major number unless we have breaking changes, which we try hard to avoid.   I believe the "shortcomings" you refer to are not shortcomings at all, but a feature.  The shortcomings are instead in the imported package, which seems not to place much value on semantic versioning.